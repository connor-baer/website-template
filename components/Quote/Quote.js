import { css } from '@emotion/core';
import styled from '@emotion/styled';

const baseStyles = ({ theme }) => css`
  margin: 1rem 0;
  padding-left: 2rem;
  font-size: 1.5rem;
  font-style: italic;
  line-height: 2.25rem;
  position: relative;

  &::before {
    display: block;
    border-top-right-radius: ${theme.borderRadius.mega};
    border-bottom-right-radius: ${theme.borderRadius.mega};
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    height: 100%;
    width: ${theme.borderRadius.mega};
    background-color: ${theme.colors.p500};
  }
`;

const Quote = styled('blockquote')(baseStyles);

export default Quote;
