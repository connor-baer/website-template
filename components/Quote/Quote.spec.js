import React from 'react';

import Quote from './Quote';

describe('Quote', () => {
  function renderQuote(fn, props = {}) {
    return fn(<Quote {...props} />);
  }

  it('should render with default styles', () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const { container } = renderQuote(render, { children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const html = renderQuote(renderToHtml, { children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
