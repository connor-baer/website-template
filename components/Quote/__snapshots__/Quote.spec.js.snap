// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render with default styles 1`] = `
.circuit-0 {
  margin: 1rem 0;
  padding-left: 2rem;
  font-size: 1.5rem;
  font-style: italic;
  line-height: 2.25rem;
  position: relative;
}

.circuit-0::before {
  display: block;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  content: '';
  position: absolute;
  left: 0;
  top: 0;
  height: 100%;
  width: 4px;
  background-color: #f06595;
}

<blockquote
  class="circuit-0 circuit-1"
>
  Bacon ipsum dolor amet swine shank biltong ham bacon.
</blockquote>
`;
